bpf/bpf_xdp.c

```diff
@@ -25,6 +25,11 @@
  */
 #define SKIP_SRV6_HANDLING
 
+/* Controls the inclusion of the CILIUM_CALL_MULTICAST_EP_DELIVERY section
+ * in the object file.
+ */
+#define SKIP_MULTICAST_EP_DELIVERY
```

Comment:
Could we use `declare_tailcall_if` to do conditionally emit this prog instead? I know it's not pretty, but we have this in our crosshairs and as long as we use the macro consistently, we can keep an overview of how many progs need this kind of behaviour.

We're looking to replace this with a system that boots unneeded progs from a CollectionSpec before loading them into the kernel to save on verification time.

Reply:
Ah, @ti-mo I'm now questioning whether this bit is needed at all. 

I was following examples of SRv6 where the tail calls are defined, not in its own specific file, but in egress_policy.h. I believe the `SKIP_SRV6_HANDLING` macro was to avoid adding SRv6 related tailcalls when only egress gateway related tailcalls are necessary (they both exist in the same egress_policy.h). 

In the multicast code tho, we segmented all multicast stuff into mcast.h, including tailcall definitions. So it should be as simple as "don't include mcast.h" to occlude the multicast tail calls from both compilation and inclusion in the prog's call map. 

Does this check out to you?

Reply:
Talked with @ti-mo a bit out of bounds. 

We may forgo the attempt at using CONFIG macro for now, but will attempt this in a follow up. 
This is partly due to the IGMP parsing method having to unroll a loop and hitting ins limits. 

The big ask here is to remove the `#define SKIP_MULTICAST_EP_DELIVERY` macro and its usage which I believe is possible without issue. Will be pushing this change shortly.

Reply:
This is now resolved, I removed the `#define SKIP_MULTICAST_EP_DELIVERY` macro and all its usages.